### PR TITLE
QR Codes

### PR DESCRIPTION
**Pull request by easoftwareGit**
Initial Draft

### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [X] All writings are my own.
- [X] My entry follows the Codecademy Docs style guide.
- [X] My changes generate no new warnings.
- [X] I have performed a self-review of my own writing and code.
- [X] I have checked my entry and corrected any misspellings.
- [X] I have made corresponding changes to the documentation if needed.
- [X] I have confirmed my changes are not being pushed from my forked `main` branch.
- [X] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [X] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by SSwiniarski** [comment]
@easoftwareGit: Two things:

- When you want the maintainers to review this you should take it out of draft.
- You need to add Metadata to the head of this entry. You can see what's needed by looking at the [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md) and the [entry template](https://github.com/Codecademy/docs/blob/main/documentation/entry-template.md). You can also see what we need by looking at the source of any other general entry.

**Comment by SSwiniarski** [comment]
@easoftwareGit Just a reminder that if you want this reviewed you need to take it out of draft.

**Comment by easoftwareGit** [comment]
I did submit another pull request, but there was no option to mark it as
not a draft.

**Comment by SSwiniarski** [comment]
@easoftwareGit I took it out of draft for you.

File: content/general/concepts/qr-code/qr-code.md
Comment (SSwiniarski) [inline content/general/concepts/qr-code/qr-code.md:1]: We're missing "Subjects" and "Tags" from the metadata.  Look at the metadata in the [entry template](https://github.com/Codecademy/docs/blob/main/documentation/entry-template.md), and then look at [subjects.md](https://github.com/Codecademy/docs/blob/main/documentation/subjects.md) and [tags.md](https://github.com/Codecademy/docs/blob/main/documentation/tags.md) to see what to list there.

File: content/general/concepts/qr-code/qr-code.md
Comment (SSwiniarski) [inline content/general/concepts/qr-code/qr-code.md:11]:
```suggestion
A **QR Code** (quick response code) is code that can be read by a computer, just like the standard bar codes that appear on products at retail stores. But the QR Code is different than the standard bar code. The standard bar code is one dimensional and reads horizontally. A QR Code is a square, with the data enclosed within the square. The three squares in the corners of the QR code tell the QR Code scanning software how the QR Code is orientated, so the scanning software can read the QR Code’s data correctly. [Here](https://www.qr-code-generator.com/qr-code-marketing/qr-codes-basics/) is a good web page detailing how the data is structured in the QR Code.  
```

File: content/general/concepts/qr-code/qr-code.md
Comment (SSwiniarski) [inline content/general/concepts/qr-code/qr-code.md:13]:
```suggestion
QR Codes are used just like their bar code counterparts: They contain data, and when scanned the data is entered into an app or is used by a device, greatly speeding up the data entry process and eliminating typing errors. Bar codes usually hold just a number, and once scanned and entered into an application, it can be used to lookup a specific item and retrieve the item’s name, price, and other information. QR Codes hold all kinds of data, so they are more widely used than bar codes.
```

File: content/general/concepts/qr-code/qr-code.md
Comment (SSwiniarski) [inline content/general/concepts/qr-code/qr-code.md:19]:
```suggestion
To read a QR Code one does not need a special laser bar code reader. Most camera apps in mobile devices contain the QR Code scanning software. Just open the camera app, focus on the QR Code, and a clickable link appears in the camera app.
```

File: content/general/concepts/qr-code/qr-code.md
Comment (SSwiniarski) [inline content/general/concepts/qr-code/qr-code.md:25]:
```suggestion
Some QR Codes are static, and some are dynamic. A static QR Code has all the data encoded, and the data will not change. If a data entry error is made during QR Code creation, that error is permanent. A dynamic QR Code fixes the problem. A dynamic QR Code will retrieve data from the internet, allowing the QR Code owner to edit the data as needed.
```

File: content/general/concepts/qr-code/qr-code.md
Comment (SSwiniarski) [inline content/general/concepts/qr-code/qr-code.md:35]: This paragraph is missing a counter-example of a QR code that would probably be unsafe to scan: A code posted on a hand-made flyer stuck to a lamppost, for example.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@easoftwareGit I've gone through and made a few comments. This seems to be a really strong entry. Sorry we didn't realize earlier that the draft issue was unintentional.

**Comment by easoftwareGit** [comment]
I signed the Contributor License Agreement

**Comment by yangc95** [comment]
> I signed the Contributor License Agreement
> […](#)

@easoftwareGit Thanks for signing the CLA! The next step would be to look over any open comments/suggestions and resolve them. Please let us know if you have any questions or need assistance.

cc: @SSwiniarski

**Comment by easoftwareGit** [comment]
I have looked over the open comments/suggestions and they all look good to me.

**Comment by SSwiniarski** [comment]
> I have looked over the open comments/suggestions and they all look good to me.

@easoftwareGit, good. As soon as you make the edits to resolve them we can go on to a second review.

**Comment by yangc95** [comment]
> > I have looked over the open comments/suggestions and they all look good to me.
> As soon as you make the edits to resolve them we can go on to a second review.

@easoftwareGit you can head over to [this PRs files](https://github.com/Codecademy/docs/pull/1343/files) and see to each comment. Suggestions can be committed separately or as a batch. See here for more info: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request

**Comment by easoftwareGit** [comment]
I have signed the agreement. I have reviewed and approved the changes. I am dyslexic, so I might be missing something I am supposed to do. I have clicked on the links to sign the Contributor License Agreement, and and completed the form. When I click on let up RECHECK it, I go to my GitHub page.

What I am not doing correctly so I can sign the license agreement?

**Comment by SSwiniarski** [comment]
> I have signed the agreement. I have reviewed and approved the changes. I am dyslexic, so I might be missing something I am supposed to do. I have clicked on the links to sign the Contributor License Agreement, and and completed the form. When I click on let up RECHECK it, I go to my GitHub page.
> 
> What I am not doing correctly so I can sign the license agreement?

@easoftwareGit, you've signed the CLA. You're fine there. You just need to go in and edit your entry to conform to the comments made above. It's your responsibility to implement the changes being suggested.

**Comment by SSwiniarski** [comment]
@easoftwareGit do you need assistance in making the changes?

**Comment by easoftwareGit** [comment]
I am getting the message "First-time contributors need a maintainer to approve running workflows. [Learn more.]
5 expected and 1 successful checks"

I have 5 mustard color dots. and each 5 have a [Required] button.  I cannot click on the [required] button. I cannot expand the messages any further. I have signed the CLA and approved  the changes. What am I missing?

**Comment by SSwiniarski** [comment]
@easoftwareGit You don't just aprove the changes. You have to go in and edit the file to make the changes that were requested. Don't worry about the workflow. We'll do that. You just have to go in and edit your file yourself to make the changes we requested.

**Comment by easoftwareGit** [comment]
Thanks for the clarification. On it!

**Comment by SSwiniarski** [comment]
Closing this PR because @easoftwareGit opened a new PR with changes, [1627](https://github.com/Codecademy/docs/pull/1627).

**Comment by easoftwareGit** [comment]
I have made the suggested changes. Thank you for your guidance and assistance, I am dyslexic, so even when I read instructions, I do not always read them correctly. Thanks you again for your patience and help.

**Pull request closed without merging**
